### PR TITLE
638 - Documentation - Add API details and improve documentation

### PR DESCRIPTION
**Pull request by paulinea**

### CODE REVIEW

File: docs/components/NavigationBottom.md [hunk 1]
```diff
@@ -19,130 +18,66 @@ description: Bottom navigation bars allow movement between primary destinations
...
-    android:layout_height="wrap_content">
-
-  <com.google.android.material.bottomnavigation.BottomNavigationView
-      android:id="@+id/bottom_navigation"
-      android:layout_width="match_parent"
-      android:layout_height="wrap_content"
-      app:menu="@menu/bottom_navigation_menu" />
-
-</LinearLayout>
+    private data class NavigationItem(
```
Comment (florentmaitre) [inline docs/components/NavigationBottom.md:41]:
```suggestion
private data class NavigationItem(
```

File: docs/components/NavigationBottom.md [hunk 2]
```diff
@@ -19,130 +18,66 @@ description: Bottom navigation bars allow movement between primary destinations
...
-In code:
+#### OdsBottomNavigation API
 
-```kotlin
-bottomNavigation.selectedItemId = R.id.page_2
-```
+Parameter | Default&nbsp;value | Description
+-- | -- | --
+`modifier: Modifier` | `Modifier` | `Modifier` applied to the bottom navigation
+`items: List<OdsBottomNavigationItem>` | | Items displayed into the bottom navigation
```
Comment (florentmaitre) [inline docs/components/NavigationBottom.md:79]: `modifier` should be below `items`, probably fixed in the other PR.

File: docs/components/NavigationDrawers.md [hunk 3]
```diff
@@ -19,26 +18,25 @@ description: The navigation drawer slides in from the left when the nav icon is
...
-        title = "Side navigation drawer" // title is mandatory in an `OdsModalDrawer`
-        imageContentDescription = "" // optional
+        title = "Side navigation drawer"
+        imageContentDescription = ""
         imageDisplayType =
             OdsModalDrawerHeaderImageDisplayType.None // or OdsModalDrawerHeaderImageDisplayType.Avatar or OdsModalDrawerHeaderImageDisplayType.Background
-        subtitle = "Example" // optional
+        subtitle = "Example"
         image =
             painterResource(id = R.drawable.placeholder)
```
Comment (florentmaitre) [inline docs/components/NavigationDrawers.md:41]:
```suggestion
        image = painterResource(id = R.drawable.placeholder)
```

File: docs/components/Sliders.md [hunk 4]
```diff
@@ -74,61 +58,46 @@ You can add icons to the continuous slider like this:
 OdsSlider(
```
Comment (florentmaitre) [inline docs/components/Sliders.md:58]: This example use the old API.

Same comment for `OdsSliderLockups` below and for discrete sliders (4 examples).

File: docs/components/AppBarsTop.md [hunk 5]
```diff
@@ -23,246 +20,104 @@ description: Top app bars display information and actions relating to the curren
...
-            onClick = { }
+            onClick = { doSomething() }
         ),
         // ...
     )
 )
 ```
 
 Note: By default, the `OdsTopAppBar` is elevated but you can set `elevated` parameter to `false` if you don't want any shadow below it (for example if you want to display tabs below).
+{:.bd-callout .bd-callout-info}
```
[the remark this answers is not in the document]
Comment (paulinea) [reply docs/components/AppBarsTop.md:68]: This was a not working test, I will remove this line ;)

File: docs/components/AppBarsTop.md [hunk 6]
```diff
@@ -23,246 +20,104 @@ description: Top app bars display information and actions relating to the curren
...
-
-    </com.google.android.material.appbar.AppBarLayout>
-
-</androidx.coordinatorlayout.widget.CoordinatorLayout>
-```
+##### OdsTopAppBar API
 
-_**Raised top app bar**_
+Parameter | Default&nbsp;value | Description
+-- | -- | --
```
[the remark this answers is not in the document]
Comment (paulinea) [reply docs/components/AppBarsTop.md:73]: Yes I think it's a good idea but maybe we can do that in another PR. What do you think?

File: docs/components/Banners.md [hunk 7]
```diff
@@ -25,42 +24,50 @@ Only one banner should be shown at a time
...
+`OdsBanner` is built to support accessibility criteria and is readable by most screen readers, such as TalkBack. The use of an `OdsBannerImage` force the developer to associate a content description to the banner image.
 
 ## Implementation
 
 ![Banner light](images/banner_light.png)
 
 ![Banner dark](images/banner_dark.png)
 
-> **Jetpack Compose implementation**
+### Jetpack Compose
```
[the remark this answers is not in the document]
Comment (paulinea) [reply docs/components/Banners.md:40]: It depends if the `Jetpack Compose` title is in a higher level title named `Implementation` or not. In this case, `implementation` word seemed to me redundant.

File: docs/components/Chips.md [hunk 8]
```diff
@@ -260,6 +221,16 @@ OdsChoiceChipsFlowRow(
...
 ```
 
+#### OdsChoiceChipsFlowRow API
+
+Parameter | Default&nbsp;value | Description
+-- | -- | --
+`value: String` | | Initial value of the choice chips flow row
+`onValueChange: (value: T) -> Unit` | | Callback invoked when the value changes. The new value is provided as parameter.
+`modifier: Modifier` | `Modifier` | `Modifier` applied to the chips flow row
+`chips: List<OdsChoiceChip<T>>` | | Chips displayed into the flow row
```
[the remark this answers is not in the document]
Comment (paulinea) [reply docs/components/Chips.md:231]: Yes this is the case in the next PR.

File: lib/src/main/java/com/orange/ods/compose/component/button/OdsButton.kt [hunk 9]
```diff
@@ -56,18 +56,16 @@ enum class OdsButtonStyle {
...
- * style on contained buttons, you can use `OdsButtonStyle.FunctionalPositive` or `OdsButtonStyle.FunctionalNegative` values.
- * @param displaySurface optional allow to force the button display on a dark or light
- * surface. By default the appearance applied is based on the system night mode value.
+ * @param text Text displayed into the button
+ * @param onClick Callback invoked when the button is clicked
+ * @param modifier [Modifier] applied to the button
+ * @param icon Icon displayed in the button before the text
+ * @param enabled Controls the enabled state of the button. When `false`, this button will not be clickable.
+ * @param style Style applied to the button. Set it to [OdsButtonStyle.Primary] for an highlighted button style or use
+ * [OdsButtonStyle.FunctionalPositive]/[OdsButtonStyle.FunctionalNegative] for a functional green/red button style.
```
[the remark this answers is not in the document]
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/compose/component/button/OdsButton.kt:65]: Yes I agree

File: docs/components/NavigationBottom.md (hunk 2 above)
Comment (paulinea) [reply docs/components/NavigationBottom.md:79]: Yes, this is the case in next PR.

File: docs/components/Chips.md [hunk 10]
```diff
@@ -15,7 +14,8 @@ description: Chips are compact elements that represent an input, attribute, or a
     * [Choice chip](#choice-chip)
     * [Filter chip](#filter-chip)
     * [Action chip](#action-chip)
-* [Modules](#modules)
+* [OdsChip API](#odschip-api)
```
[the remark this answers is not in the document]
Comment (paulinea) [reply docs/components/Chips.md:17]: There is a problem with that point as sometimes we have 2 or more paragraphs called `Jetpack Compose implementation` in the same page. In this case it is difficult to add correct links to it. I found workarounds but I don't know which one to use: [https://stackoverflow.com/questions/57935181/markdown-anchor-link-with-same-name-but-different-sections](https://stackoverflow.com/questions/57935181/markdown-anchor-link-with-same-name-but-different-sections)

Any idea?

File: docs/components/AppBarsTop.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (florentmaitre) [reply docs/components/AppBarsTop.md:71]: Yes, I did not mention it but that's what I was thinking about 😃

**Review by florentmaitre — APPROVED** [review]

**Review by florentmaitre — APPROVED** [review]

**Pull request merged**